src/cpp/src/prompt_lookup/continuous_batching_for_prompt_lookup.cpp

```diff
@@ -67,7 +67,7 @@ void ContinuousBatchingPipeline::ContinuousBatchingForPromptLookupImpl::generate
             const auto sampling_params = request->get_sampling_parameters();
             {
                 const auto generated_len = running_sequence->get_generated_len();
-                const auto left_generated_len = std::min(sampling_params.max_new_tokens, sampling_params.max_length) - generated_len - 1;
+                const auto left_generated_len = std::min(request->get_max_new_tokens(), sampling_params.max_length) - generated_len - 1;
```

Comment:
I suppose it can be incorrect to take min between `sampling_params.max_length` and `request->get_max_new_tokens()`, because first one assumes prompt 

@iefode should not we use just `request->get_max_new_tokens()` ?

Reply:
@ilya-lavrenov Correct, thanks!

Looks like we should use:
`const auto left_generated_len = std::min(request->get_max_new_tokens(), sampling_params.max_length - prompt_len) - generated_len - 1;`

Reply:
`request`s get_max_new_tokens:
```c++
size_t get_max_new_tokens() const {
    return m_sampling_params.get_max_new_tokens(get_prompt_len());
}
```

`m_sampling_params`s get_max_new_tokens:
```c++
size_t GenerationConfig::get_max_new_tokens(size_t prompt_length) const {
    // max_new_tokens has priority over max_length, only if max_new_tokens was not specified use max_length
    if (max_new_tokens != SIZE_MAX) {
        return max_new_tokens;
    } else {
        return max_length - prompt_length;
    }
}
```

It looks we can use just `request->get_max_new_tokens()`

Reply:
@michalkulakowski please, take into account details above from Alexander